tests/test_auto_offer_on_device_rest.py

```diff
@@ -43,24 +58,33 @@ def on_trade(self, trade_info):
     def on_finish(self, finish_info):
         self.is_finished = True
 
-number_of_areas = 10
-load_names = [str(sys.argv[2]) if i == 0 else str(sys.argv[2])+' '+str(i+1) for i in range(number_of_areas)]
-pv_names = [str(sys.argv[3]) if i == 0 else str(sys.argv[3])+' '+str(i+1) for i in range(number_of_areas)]
-connected_devices = []
+#create a list with the number of areas
+areas=range(number_of_areas)
+#split the list to have one list for each researcher
+#the list contains lists of the devices numbers each researcher will connect to
+areas_split = [i.tolist() for i in np.array_split(np.array(areas),number_of_researcher)]
+
+if args.researcher_id <= number_of_researcher:
+    #create a list with the names of the Load the researcher connects to 
+    load_names = [args.load if i == 0 else f'{args.load} {i+1}' for i in areas_split[int(args.researcher_id-1)]]
+    #create a list with the names of the PV the researcher connects to 
+    pv_names = [args.pv if i == 0 else f'{args.pv} {i+1}' for i in areas_split[int(args.researcher_id-1)]]
+else:
+    raise ValueError(f'The researcher id is out of range. The number of researcher is {number_of_researcher}.')
```

Comment:
I think i understand what you are trying to do here but IMO this could be done simpler.
How about not having consecutive area ids/names but just add the researcher id in the area_name
for example something like this:
`pv_names = [f"area_{researcher_id}_{area_number}" for researcher_id in range(len(number_of_researcher)) for area_number in number_of_areas ]`
Or did i get it wrong?
It would be good if you could omit using numpy because it is a large library, takes long time to install (needs to be done every time when the integrationtests are running on travis). Especially it you only need it for list/array mutations.
Thanks!

Reply:
I actually split it that way because the API gets the device uuid with the device name. So the name refers to the one given by default when I create the area on the JS test: 'Load, Load 2 ...'. If I use the researcher_id in the names, it implies I have to rename in the test all devices when they are created (correct me if I'm wrong).
Therefore, I don't know what it is better: split in the API script based on the default name, or rename one by one the devices in the test ?

Reply:
Also, does the number of researcher should be "hard-coded" ? For example, if we decide to have the test running with always 2 researchers then I don't need numpy and I can directly attribute the first 5 devices names to the first researcher and the 5 next to the other (or rename the devices with researcher_id 1 or 2 in the collaboration). 
I used numpy in order to split the devices accordingly to the number of researcher, I am wondering is this necessary or not ? (I also guess I could do it without numpy)